engine/src/main/java/org/destinationsol/game/planet/SolarSystemConfig.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2021 The Terasology Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.destinationsol.game.planet;
+
+import org.destinationsol.game.ShipConfig;
+import org.destinationsol.game.chunk.SpaceEnvConfig;
+import org.destinationsol.game.item.TradeConfig;
+
+import java.util.ArrayList;
+
+public class SolarSystemConfig {
+
+    public final String name;
+    public final ArrayList<ShipConfig> tempEnemies;
+    public final SpaceEnvConfig envConfig;
+    public final ArrayList<ShipConfig> constEnemies;
+    public final ArrayList<ShipConfig> constAllies;
+    public final TradeConfig tradeConfig;
+    public final ArrayList<ShipConfig> innerTempEnemies;
+    public final boolean hard;
+
+    public SolarSystemConfig(String name, ArrayList<ShipConfig> tempEnemies, SpaceEnvConfig envConfig,
```

Comment:
I wonder if the Builder pattern might be better here, to avoid having a bunch of parameters listed like this

Reply:
@IsaacLic Turns out most of the parameters are pretty useless, as the are only used to pass in empty arrays or objects, and then those are set from JSON data by the load() method. I'm changing it so that those empty objects are initialized in the constructor instead of being passed in